documentation/library_guide/kernel_templates/esimd/radix_sort.rst

```diff
@@ -0,0 +1,322 @@
+         std::cout << '\n';
+
+         sycl::host_accessor v_acc{values, sycl::read_only};
+         for(std::size_t i = 0; i < n; ++i)
+               std::cout << v_acc[i] << ' ';
+         std::cout << '\n';
+      }
+
+      return 0;
+   }
+
+**Output:**
+
+.. code:: none
+
+   1 2 3 3 3 5
+   s o r t e d
+
+
+-----------------------------------------
+Recommended Settings for Best Performance
+-----------------------------------------
+
+The general advice is to set your configuration according to the performance measurements and profiling information. The initial configuration may be selected according to these points:
+
+- When the number of elements to sort is small (~16K or less) and the algorithm is ``radix_sort``, then the elements can be processed by a single work-group. Increase the param values, so ``N <= param.data_per_workitem * param.workgroup_size``.
+
+- When the number of elements to sort is medium (between ~16K and ~1M), then all the work-groups can execute simultaneously. Make sure the device is saturated: ``param.data_per_workitem * param.workgroup_size ≈ N / device_xe_core_count``. A larger ``param.workgroup_size`` in ``param.data_per_workitem * param.workgroup_size`` combination is preferred to reduce the number of work-groups and the synchronization overhead.
+
+- When the number of elements to sort is large (more than ~1M), then the work-groups preempt each other. Increase the occupancy to hide the latency with ``param.data_per_workitem * param.workgroup_size ≈< N / (device_xe_core_count * desired_occupancy)``. The occupancy depends on the local memory usage, which is determined by ``key_type``, ``value_type``, ``radix_bits``, ``param.data_per_workitem`` and ``param.workgroup_size`` parameters. Refer to :ref:`Local Memory <local-memory>` section for the calculation.
+
+
+.. _limitations:
+
+-----------
+Limitations
+-----------
+
+- Algorithms can only process C++ integral and floating-point types with a width of up to 64 bits (except for ``bool``).
+- Number of elements to sort must not exceed `2^30`.
+- ``radix_bits`` can only be `8`.
+- ``param.data_per_workitem`` has discreteness of `32`.
+- ``param.workgroup_size`` can only be `64`.
+- Local memory is always used to rank keys, reorder keys, or key-value pairs, which limits possible values of ``param.data_per_workitem`` and ``param.workgroup_size``
+- ``radix_sort_by_key`` does not have single-work-group implementation yet.
+
+
+.. _possible-api-extensions:
+
+-----------------------
+Possible API Extensions
+-----------------------
+
+- Allow passing externally allocated memory.
+- Allow passing dependent events.
+- Allow passing a range of bits to sort.
+- Allow out-of-place sorting (for example, through a double-buffer or an output sequence).
+- Allow configuration of kernels other than the most time-consuming kernel (for example, of a kernel computing histograms).
+- Allow range transformations (for example, range pipes or transform iterators).
```

Comment:
If we include this section, should we provide some method for users to provide feedback on which possible / future extensions would be most important to them?

Otherwise, I'm not sure how relevant this section is to be user-facing.  As far as I know, they are not concrete enough plans for users to "plan" around.  I do think getting feedback here could be valuable though.

Reply:
I read this as these APIs are currently available, but I agree it should be removed if this is just a list of what could be.

Reply:
As far as I understand, it is a list of possible future avenues of extension which do not currently exist.  Since this is experimental, the API may shift as we add features.

Reply:
If we don't already have the APIs, I don't think we should highlight them.

Reply:
I've removed that section. 

As for the feedback, I think, users can open an issue. I can create a note for it.

Reply:
If you do add a note, please direct users to the Contributing page: https://github.com/oneapi-src/oneDPL/blob/release_oneDPL/CONTRIBUTING.md